### DISCUSSION TITLE
How can I properly remove etcd masters to reduce my cluster to a single master?

### DISCUSSION THREAD

**Discussion by 2fst4u** [k3s-io/k3s#6019] [Q&A] [answered]
I have a cluster that had 3 masters with etcd. The cluster is made up of raspberry pis and was working ok until a recent update made etcd unstable and the masters would continually drop out of readiness, screwing the quorum and wreaking havoc on my longhorn volumes.

I want to reduce my masters to one single master, and if also needed, make it unschedulable so it just has to focus on being a master.

I successfully deleted one of the masters, uninstalled k3s and re-added it as a worker. This was fine.

When I deleted the second master, the remaining master stopped working completely and I had no control plane.

After much troubleshooting, I attempted to re-add the second master and it made both that one, and the non-working remaining master start working.

I've just read another topic that said this is because etcd loses quorum which brings down the control plane.

Now to my question:

How can I properly remove the two masters I don't want and reduce etcd to just a single master without this issue? I was under the impression `kubectl delete node` would gracefully do everything needed but apparently not.

**Comment by brandond** [comment, marked as answer]
`kubectl delete node` will remove a node from the cluster, but only if the cluster is currently functional. If it is not, you can start one of the servers with the `--cluster-reset` flag to reset the etcd cluster to a single node. Note that this will not delete the other nodes from the Kubernetes cluster; it just resets the etcd cluster itself.

**Reply by 2fst4u** [reply]
Interesting to note. Couldn't find much information on that flag so I wasn't sure of its use.

I tried the regular node delete again and It seems to have worked without issue. I think the previous effort was interrupted but the cluster flapping up and down, and the second node wasn't actually deleted properly. This time I stopped everything and calmed down the cluster before trying and it seems stable.
